src/hotspot/share/logging/logDecorations.cpp

```diff
@@ -30,17 +30,17 @@
 #include "runtime/os.hpp"
 #include "services/management.hpp"
 
-const char* volatile LogDecorations::_host_name = NULL;
+const char* volatile LogDecorations::_host_name = nullptr;
 const int LogDecorations::_pid = os::current_process_id(); // This is safe to call during dynamic initialization.
 
 const char* LogDecorations::host_name() {
   const char* host_name = Atomic::load_acquire(&_host_name);
-  if (host_name == NULL) {
+  if (host_name == nullptr) {
     char buffer[1024];
     if (os::get_host_name(buffer, sizeof(buffer))) {
       host_name = os::strdup_check_oom(buffer);
-      const char* old_value = Atomic::cmpxchg(&_host_name, (const char*)NULL, host_name);
-      if (old_value != NULL) {
+      const char* old_value = Atomic::cmpxchg(&_host_name, (const char*)nullptr, host_name);
```

Comment:
I wonder if you need this cast with nullptr ?

Reply:
The nullptr argument is templated, so we do need the cast.

Reply:
Thanks for checking.  I won't ask on the next one. :)

Reply:
I'd like to add overloads for std::nullptr_t someday.